src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2793,14 +2793,24 @@ def update_linux_profile(self, mc: ManagedCluster) -> ManagedCluster:
 
         if ssh_key_value:
             if mc.linux_profile is None:
-                raise InvalidArgumentValueError("Updating the cluster with no ssh key set at creation is not supported")
-            mc.linux_profile.ssh = self.models.ContainerServiceSshConfiguration(
-                public_keys=[
-                    self.models.ContainerServiceSshPublicKey(
-                        key_data=ssh_key_value
+                mc.linux_profile = self.models.ContainerServiceLinuxProfile(
+                    admin_username="azureuser",
+                    ssh=self.models.ContainerServiceSshConfiguration(
+                        public_keys=[
+                            self.models.ContainerServiceSshPublicKey(
+                                key_data=ssh_key_value
+                            )
+                        ]
                     )
-                ]
-            )
+                )
+            else:
+                mc.linux_profile.ssh = self.models.ContainerServiceSshConfiguration(
```

Comment:
still need to check if .ssh is None or not?

Reply:
no matter it's none or not, we need to update `.ssh`.